src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -348,32 +348,13 @@ private async Task<bool> IsValidAuthTypeAsync(User user, string grantType)
             return true;
         }
 
-        // Is user apart of any orgs? Use cache for initial checks.
-        var orgs = (await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, user.Id))
-            .ToList();
-        if (orgs.Any())
+        // Check if user belongs to any organization with an active SSO policy 
+        // If 'EnforceSsoPolicyForAllUsers' is set to true then SSO policy applies to all user types otherwise it does not apply to Owner or Admin
+        var minSsoPolicyUserType = _globalSettings.Sso.EnforceSsoPolicyForAllUsers ? OrganizationUserType.Owner : OrganizationUserType.User;
```

Comment:
This is an example of logic that should be within the `PolicyService` method.

Reply:
This is now part of `PolicyService`.